### PR TITLE
Fix #41 : rm holdSearch non sensible API

### LINKED ISSUE #41: Remove holdSearch parameter!
See #36 for complete discussion.

Expected API  : 

```javascript
helper.setDisjunctiveRefine( "facet1", "value" )
      .setQuery( "some input" )
      .search();
```

 - [x] Remove holdsearch parameter
 - [x] Remove autosearch magic
 - [x] Create a setQuery function and let search be the method that executes the search
 - [x] Make mutation API chainable
 - [x] Update tests
 - [x] Update samples

### PR DESCRIPTION
**Pull request by bobylito**
In order to clarify the API for the users, this pull requests changes the following : 
 - no automatic search on state changes
 - no more hold search parameter to prevent this behavior
 - chainable mutation api to ease the use
 - separate genuine search from setting the query

### CODE REVIEW

**Comment by bobylito** [comment]
@redox @vvo Ping :)

File: src/algoliasearch.helper.js [hunk 1]
```diff
@@ -22,60 +22,60 @@ function AlgoliaSearchHelper( client, index, options ) {
...
- * Perform a query
+ * Sets the query. Also sets the current page to 0.
  * @param  {string} q the user query
- * @param {boolean} holdSearch Optionnal parameter to specify that the search should not be triggered right away
+ * @return {AlgoliaSearchHelper}
  */
-AlgoliaSearchHelper.prototype.search = function( q, holdSearch ) {
+AlgoliaSearchHelper.prototype.setQuery = function( q ) {
   this.state = this.state.setQuery( q )
                          .setPage( 0 );
```
Comment (redox) [inline src/algoliasearch.helper.js]: Are we really returning the state here? I assume we're returning `this`, isn't it?

Comment (bobylito) [inline src/algoliasearch.helper.js]: this.state is a SearchParameter. And it returns SearchParameter when mutated. The chainable part the helper is handled a few lines later https://github.com/algolia/algoliasearch-helper-js/pull/43/files#diff-cbf795d639d5f4b610b6d0f6ee395b2aR44

Comment (redox) [inline src/algoliasearch.helper.js]: Ah yes, my bad.

**Comment by vvo** [comment]
[LGTM](http://www.lgtm.in/i/OpwrWn9Sp)

**Pull request merged**